### PR TITLE
fixes for -Wextra -Wpedantic -Wshadow -Wconversion 

### PR DESCRIPTION
**Pull request by junghans**
https://github.com/votca/votca/pull/212

### CODE REVIEW

File: include/votca/tools/akimaspline.h [hunk 1]
```diff
@@ -65,7 +63,7 @@ class AkimaSpline : public Spline {
 
   // fit spline through noisy data
   // x,y are arrays with noisy data, both vectors must be of same size
-  void Fit(Eigen::VectorXd &x, Eigen::VectorXd &y) override;
```
Comment (JensWehner) [inline include/votca/tools/akimaspline.h]: I do not understand this change?

Comment (junghans) [reply include/votca/tools/akimaspline.h]: x and y aren’t used in the function, so making them anonymous.

**Comment by junghans** [comment]
From me!

File: include/votca/tools/graphnode.h [hunk 2]
```diff
@@ -71,8 +70,6 @@ class GraphNode {
   /// Get the string id unique to the contents of the graph node
   std::string getStringId() const { return str_id_; }
 
-  GraphNode& operator=(const GraphNode& gn);
```
Comment (junghans) [inline include/votca/tools/graphnode.h]: @JensWehner please check

File: include/votca/tools/graphvisitor.h [hunk 3]
```diff
@@ -50,32 +50,31 @@ class Graph;
...
-  std::set<int> explored_;
+  std::set<long int> explored_;
 
   /// The vertex the visitor started on
-  int startingVertex_ = 0;
+  long int startingVertex_ = 0;
 
   /// What is done to an individual graph node as it is explored
-  virtual void addEdges_(const Graph& graph, int vertex) = 0;
-  virtual Edge getEdge_(const Graph& graph) = 0;
```
Comment (junghans) [inline include/votca/tools/graphvisitor.h]: @JoshuaSBrown, not 100% sure here, but `graph` was never used anywhere.

File: include/votca/tools/histogramnew.h [hunk 4]
```diff
@@ -161,7 +161,7 @@ class HistogramNew {
   void setPeriodic(bool periodic) { _periodic = periodic; }
 
  private:
-  void Initialize_(double min, double max);
+  void Initialize_();
```
Comment (junghans) [inline include/votca/tools/histogramnew.h]: @JensWehner this was just misleading as never used.

File: src/libtools/graphvisitor.cc [hunk 5]
```diff
@@ -33,13 +33,14 @@ class GraphNode;
 
 bool GraphVisitor::queEmpty() const { return true; }
 
-void GraphVisitor::exploreNode(pair<int, GraphNode>& vertex_and_node,
-                               Graph& graph, Edge edge) {
+void GraphVisitor::exploreNode(pair<long int, GraphNode>& vertex_and_node,
+                               __attribute__((unused)) Graph& graph,
+                               __attribute__((unused)) Edge edge) {
```
Comment (junghans) [inline src/libtools/graphvisitor.cc]: @JoshuaSBrown not sure if there is a problem in the design here.

File: src/libtools/property.cc [hunk 6]
```diff
@@ -466,6 +469,7 @@ std::ostream &operator<<(std::ostream &out, const Property &p) {
     switch (type) {
       default:
         PrintNodeTXT(out, p, level);
+        break;
```
Comment (junghans) [inline src/libtools/property.cc]: @JensWehner hopefully we never used txt output anywhere....

File: src/tests/test_graphvisitor.cc [hunk 7]
```diff
@@ -32,12 +32,10 @@ using namespace votca::tools;
...
 class GraphVisitorTest : public GraphVisitor {
  private:
-  void addEdges_(const Graph&, int) override {
-    throw runtime_error("Undefined method.");
-  }
-  Edge getEdge_(const Graph&) override {
+  void addEdges_(const Graph&, long int) override {
     throw runtime_error("Undefined method.");
   }
+  Edge getEdge_() override { throw runtime_error("Undefined method."); }
```
Comment (junghans) [inline src/tests/test_graphvisitor.cc]: @JensWehner maybe `Edge getEdge_() = delete;`

File: src/libtools/property.cc (hunk 6 above)
Comment (JensWehner) [reply src/libtools/property.cc]: do not think so.

File: src/tests/test_graphvisitor.cc (hunk 7 above)
Comment (JensWehner) [reply src/tests/test_graphvisitor.cc]: yeah for all of them

**Comment by junghans** [comment]
@JensWehner can you have a look at the two codacy warnings?

**Pull request merged**